addon/compression_worker.h

```diff
@@ -0,0 +1,73 @@
+#ifndef COMPRESSION_WORKER_H
+#define COMPRESSION_WORKER_H
```

Comment:
I'd namespace this #define, e.g. 

```suggestion
#ifndef MONGODB_ZSTD_COMPRESSION_WORKER_H
#define MONGODB_ZSTD_COMPRESSION_WORKER_H
```

since this is effectively just a single global namespace